threedigrid_builder/grid/cross_section_definitions.py

```diff
@@ -43,36 +49,47 @@ def convert(self, ids):
             content_pk=ids,
             code=self.code[idx],
             count=0,
+            count_yz = 0,
         )
         if len(result) == 0:
             return result
 
-        offset = 0
         tables = []
+        tables_yz = []
+
 
         # Numpy array views on width/height based on idx
         width_idx = self.width[idx]
         height_idx = self.height[idx]
 
         for i, shape in enumerate(self.shape[idx]):
             tabulator = tabulators[shape]
-            result.shape[i], result.width_1d[i], result.height_1d[i], table = tabulator(
+            result.shape[i], result.width_1d[i], result.height_1d[i], table, yz = tabulator(
                 shape, width_idx[i], height_idx[i]
             )
             if table is not None:
                 result.count[i] = len(table)
-                result.offset[i] = offset
-                offset += len(table)
                 tables.append(table)
+            if yz is not None:
+                result.count_yz[i] = len(yz)
+                tables_yz.append(yz)
```

Comment:
I suggest inserting the friction value over here in a specialized function `set_vegetation_friction_values(yz, friction_values, vegetation_stem_densities, vegetation_stem_diameters, vegetation_heights, vegetation_drag_coefficients)`